### ISSUE TITLE
when install bun, faced crash 

### ISSUE THREAD

**Issue by glulion** [oven-sh/bun#10928]
### How can we reproduce the crash?

bun install

![image](https://github.com/oven-sh/bun/assets/81935266/67a71ba7-52a1-436d-ae64-cf5aa9bb5542)

### JavaScript/TypeScript code that reproduces the crash?

```shell
bun install v1.1.7 (b0b7db5c)
  🚚 turbo [127/129] ============================================================
Bun v1.1.7 (b0b7db5c) Windows x64 (baseline)
Args: "C:\Program Files\nodejs\node_modules\bun\bin\bun.exe", "install"
Elapsed: 30727ms | User: 296ms | Sys: 593ms
RSS: 0.18GB | Peak: 0.21GB | Commit: 0.26GB | Faults: 58213

panic(thread 8748): Illegal instruction at address 0x7FF6AC20FA41
oh no: Bun has crashed. This indicates a bug in Bun, not your code.

To send a redacted crash report to Bun's team,
please file a GitHub issue using the link below:
```

### Stack Trace (bun.report)

Bun v1.1.7 ([`b0b7db5`](https://github.com/oven-sh/bun/tree/b0b7db5c060e830801b0273b7ff6e34e3eb5c254)) on windows x86_64_baseline [InstallCommand]

Illegal instruction at address 0x7FF7D672FA41

- [`libtcc.c:188`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/deps/tinycc/libtcc.c#L188): `pstrcpy`
- [`tccgen.c:5018`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/deps/tinycc/tccgen.c#L5018): `post_type`
- [`logger.zig:993`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/logger.zig#L993): `addRangeErrorFmt`
- [`semver.zig:268`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/install/semver.zig#L268): `slice`
- [`math.zig:452`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/deps/zig/lib/std/math.zig#L452): `sub`
- [`bun.zig:1754`](https://github.com/oven-sh/bun/blob/b0b7db5c060e830801b0273b7ff6e34e3eb5c254/src/bun.zig#L1754): `insert`
- ??? at `0x17344` in KERNEL32.DLL
- ??? at `0x526b1` in ntdll.dll

**Comment by paperclover** [comment]
this was likely fixed in #10912. can you try `bun upgrade --canary` and try this again?

**Comment by glulion** [comment]
> this was likely fixed in #10912. can you try `bun upgrade --canary` and try this again?

Thanks it is fixed.
